[Lim Wei Hern] iP

nreHieW opened the pull request:
## James
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

James is a helpful chatbot assistant that helps you keep track of your tasks. It is 
* text-based
* easy to learn 
* ~~fast~~ *SUPER FAST* to learn and get started 

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html)
2. double-click it to start
3. add your tasks
4. let it manage your tasks for you 😄 

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
    public static void main(String[] args) {
        new James("data/James.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

nreHieW committed:
Rename to James

nreHieW committed:
Added User Input and Echo Functionality

nreHieW committed:
Add ArrayList to Store Tasks with

nreHieW committed:
Add Task Functionality

nreHieW committed:
Add Different Task

nreHieW committed:
Add Automated Test

nreHieW committed:
Add Error Handling

nreHieW committed:
Add Error Handling

nreHieW committed:
Add Delete Functionality

nreHieW committed:
Add Enums and move main logic out of main

nreHieW committed:
Renamed Child Task Files

nreHieW committed:
Convert items to instance attribute

nreHieW committed:
Rename Task files and implement saving

nreHieW committed:
Change save file path

nreHieW committed:
Add data directory to gitignore

nreHieW committed:
Merge branch 'branch-Level-7'

nreHieW committed:
Include DateTime parsing

nreHieW committed:
Update test cases

nreHieW committed:
Merge branch 'branch-Level-8'

nreHieW committed:
Implement more classes

nreHieW committed:
Update test case

nreHieW committed:
Merge branch 'branch-A-MoreOOP'

nreHieW committed:
Refactor into packages

nreHieW committed:
Merge branch 'add-gradle-support'

nreHieW committed:
Update gradle run classname

nreHieW committed:
Add CheckStyle

nreHieW committed:
Refactor directory structure

nreHieW committed:
Add JUnit tests for Parser and TaskList

nreHieW committed:
Create Jar

nreHieW committed:
Added Docstrings

nreHieW committed:
Fix style issues based on checkstyle

nreHieW committed:
Fix style issues based on checkstyle

nreHieW committed:
Merge branch 'branch-A-Javadoc'

nreHieW committed:
Add find functionality

nreHieW committed:
Merge branch 'branch-A-CodingStandard'

nreHieW committed:
Merge branch 'branch-Level-9'

nreHieW committed:
Add Javadoc for Find

nreHieW committed:
Edit Readme

nreHieW committed:
Refactor package

nreHieW committed:
Add UI for James

nreHieW committed:
Fix checkstyle errors

nreHieW committed:
Merge branch 'branch-Level-10'

revdrag commented on src/main/java/james/James.java:
```diff
@@ -0,0 +1,79 @@
+            ui.showLoadingError();
+            this.tasks = new TaskList();
+        }
+    }
```
Perhaps it would be good to separate methods with a blank line?

revdrag commented on src/test/java/james/ParserTest.java:
```diff
@@ -0,0 +1,54 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
```
I like how you segment the different types of the import statements.

revdrag commented on src/test/java/james/TaskListTest.java:
```diff
@@ -0,0 +1,29 @@
+package james;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Perhaps it would be good to follow the same logic for all files in segmentation of import statements?

revdrag commented on src/main/java/james/Ui.java:
```diff
@@ -0,0 +1,55 @@
+import java.util.Scanner;
+
+/**
+ * Represents the user interface of the program.
```
I like how you included JavaDoc for the class itself as well and not only the methods! 🤩

revdrag commented on src/main/java/james/Ui.java:
```diff
@@ -0,0 +1,55 @@
+    private static String line = "_________________________";
+
+    /**
+     * The parser used to parse user input.
```
Maybe you can leave this JavaDoc comment as a single line since it is a field?

revdrag reviewed (commented):
Nice and easy to read! 😃 Some nitpicking and it is probably IntelliJ with their additional Git tags on the side but there are multiple classes with more than 1 blank line.

nreHieW committed:
Added Assertions to various tasks

This commit adds assertions to check if the time input for both DeadlineTask and EventTask is present.
It also checks if the input to James::processInput is null.

In sum, this checks if the parser has accurately parsed the user input.

nreHieW committed:
Improved Code Quality of James

This commit improved the readability of the messages from James after each command. Previously, the strings were added but now
they are written in a single line which helps make it more readable.

This commit also includes comments for the different types of commands to make the parsing easier to understand.

nreHieW committed:
Merge pull request #2 from nreHieW/branch-A-CodeQuality

Improve Code Quality of James

nreHieW committed:
Merge pull request #1 from nreHieW/branch-A-Assertions

Add Assertions to various tasks

nreHieW committed:
Add github CI

nreHieW committed:
Merge pull request #3 from nreHieW/branch-A-CI

Add github CI

nreHieW committed:
Add check for duplicate tasks

This commit adds check for duplicate tasks (Extension C-DetectDuplicates). The check is by checking the tostring() method

nreHieW committed:
Update tests

nreHieW committed:
Merge pull request #4 from nreHieW/branch-BCD-extension

Add check for duplicate tasks

nreHieW committed:
Add User Guide

nreHieW committed:
Merge pull request #6 from nreHieW/branch-A-UserGuide

Add User Guide

nreHieW committed:
Rename Ui.png
